modules/core/test/src/smithy4s/http/internals/PathParsingSpec.scala

```diff
@@ -0,0 +1,36 @@
+/*
+ *  Copyright 2021 Disney Streaming
+ *
+ *  Licensed under the Tomorrow Open Source Technology License, Version 1.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *     https://disneystreaming.github.io/TOST-1.0.txt
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package smithy4s.http.internals
+
+import smithy4s.http.PathSegment
+
+object PathParsingSpec extends weaver.FunSuite {
+
+  test("Parse path pattern into path segments") {
```

Comment:
I think it's worth having a unit test that verifies how a specific operation encodes to a HttpEndpoint. And maybe a couple examples for paths alone

Reply:
For sure. Not sure I'll have much time this week (I already spent much more time on this today than I should have)